docs/design/virtual-addresses.md

```diff
@@ -0,0 +1,123 @@
+- Introduce new `CryptoGetAccountVirtualAddressesQuery` in order to return unbounded list of virtual addresses for an account
+
+### Global properties
+Create the following properties to configure the virtual addresses' behavior:
+- `virtualAddresses.maxNumber` - the maximum number of virtual addresses per account
+- `virtualAddresses.canDisable` - toggle for the ability to disable virtual accounts
+- `virtualAddresses.canRemove` - toggle for the ability to remove virtual accounts
+
+## Phases of development
+
+The development will be done in iterative phases that build on previous ones. Protobuf changes needed for a given phase should be introduced in the preceding phase.
+
+- Phase 0
+  - Protobuf changes:
+    - Add virtual addresses list to `AccountInfo` proto (already added)
+  - Create design doc and test plan
+- Phase 1
+  - Implement changes to state
+    - Add virtual addresses list account state (capped to 1)
+    - Track nonces by `evmAddress` and support incrementing a specific address increment
+  - Account aliases map split
+    - Introduce the `evmAddress -> accountId` map
+    - Update alias resolution logic to work with `alias -> accountId` map for public key aliases and with `evmAddress -> accountId` map for `evmAddress` aliases
+    - Add support for `CryptoGetInfoQuery` with `evmAddress` to use the `evmAddress -> accountId` map
+  - Create new system account for exposed keys from Ethereum tools and external chains
+- Phase 2
+  - Protobuf changes
+    - Add `virtual_address_override` to `ContractCall` and `ContractCreate` transactions
+  - Account migration
+    - All ECDSA accounts with an alias get a single virtual address
+    - All ECDSA accounts with `evmAddress` stored in alias path get a single virtual address 
+    - All ECDSA accounts which calculated addresses are not present in the map will get a single virtual address
+    - All contracts with valid 20 byte addresses will get a single virtual address
+  - Accounts to utilize virtual addresses on creation
+    - EOAs set virtual address for regular `CryptoCreate` with ECDSA key, auto-create with alias and lazy-crate with `evmAddress`
+    - Contracts set virtual address for Smart contract `new()` (`CREATE` & `CREATE2`) and `EthereumTransaction CREATE` with empty `to` address
+    - Creation logic verifies `evmAddress` uniqueness and fails/doesn't set if entry already exists
+- Phase 3
+  - Protobuf changes
+    - Add disable to `CryptoUpdateTransactionBody.virtual_address_update`
+  - Expand virtual address limit per account to 3
+  - Virtual address addition to existing accounts with signature verification logic and verification for `evmAddress` uniqueness
+  - Support `virtual_address_override` logic for `ContractCreate` and `ContractCall` transaction
+- Phase 4
+  - Support virtual address disabling on `CryptoUpdate`
+  - Design virtual address deletion & transfer
+
+## Acceptance Tests
+
+### Positive Tests
```

Comment:
There should be good tests - pos & neg - specifically relating to the _migration_ (which is not reversible in production).

Reply:
(still don't see migration tests in the split-out test-plans/hip-631.md)

Reply:
Added separate section for migration tests in the test plan doc.